tensorflow/python/keras/utils/vis_utils_test.py

```diff
@@ -103,6 +104,167 @@ def test_plot_model_with_add_loss(self):
     except ImportError:
       pass
 
+  def test_dot_layer_range(self):
+    model = efficientnet.EfficientNetB0()
+
+    # Case 1: When layer_range has indexes
+    layer_range = [9, 29]
```

Comment:
By reading the test code here, I think the int input for the layer range is very hard to reason about, and also very fragile for the model network change. Even with the same layers, the topology change when branching and merging could cause this to break.

I think we should focus on the string names and maybe add regex string name to match more layers.

Reply:
@qlzh727 , I have made changes and now `layer_range` only expects strings which will be layer names.

Although, I couldn't understand what you said for matching with regex. Can you please elaborate more on that and if possible provide an example?

Reply:
The efficientnet as example, it has a bunch of blocks that has same structure, and they just have name with different suffix number. User could choose to plot specific block by "block-0" or "block-1", or all the same blocks by "block-*"

Reply:
@qlzh727 , In that case, the subgraphs would be distinct and what should we consider as subgraph to plot? If let's say user passed [block-*, prediction], then should we plot from starting of first block to last layer? Since the first element of layer_range in this case is pattern which exists at different positions in model graph.

Reply:
yea, the start predicate will be the first element it matches, and the end predicate will be the last element it matches.